app/models/repository.rb

```diff
@@ -321,27 +323,25 @@ def self.create_from_data(repo_host_data)
     nil
   end
 
-  def check_status
-    Repository.check_status(host_type, full_name)
-  end
-
   # @return [Boolean] false if we received a 404 response code when checking the status, true otherwise
-  def self.check_status(host_type, repo_full_name)
+  def check_status
     domain = RepositoryHost::Base.domain(host_type)
-    response = Typhoeus.head("#{domain}/#{repo_full_name}")
+    response = Typhoeus.head("#{domain}/#{full_name}")
+
+    return true if status == "Hidden" # don't overwrite Hidden projects
 
     if response.response_code == 404
-      repo = Repository.includes(:projects).find_by_full_name(repo_full_name)
-      if repo
-        repo.update_attribute(:status, "Removed") unless repo.private?
-        repo.projects.each do |project|
-          next unless %w[bower go elm alcatraz julia nimble].include?(project.platform.downcase)
-
-          project.update_attribute(:status, "Removed")
-        end
+      update(status: "Removed", status_reason: "Response 404") unless private?
+
+      projects.each do |project|
+        next unless %w[bower go elm alcatraz julia nimble].include?(project.platform.downcase)
+
+        project.update_attribute(status: "Removed", status_reason: "Response 404")
       end
 
       return false
+    elsif response.response_code == 200
+      update(status: nil, status_reason: "Response 200")
     end
```

Comment:
@wenottingham I think this might fix the cases you noticed, where the Repository is "Removed" but the repository link is actually working, e.g.:

<img width="1246" alt="Screenshot 2024-08-14 at 1 08 22 PM" src="https://github.com/user-attachments/assets/80e5171c-de45-4096-91b8-be992b3f8c86">